### ISSUE TITLE
Enter confirm the row edit without validation process

### ISSUE THREAD

**Issue by IniterWorker** [material-table-core/core#390]
**Describe the bug**

We have two paths to "Check" a row in edit mode. But both are different: 

- [m-table-edit-row.js](https://github.com/material-table-core/core/blob/master/src/components/MTableEditRow/m-table-edit-row.js) 
- First path is from "actions" icon in the row.

```js
const isValid = this.props.columns.every((column) => {
      const error = validateInput(column, this.state.data);
      return error.isValid;
    });
    const actions = [
      {
        icon: this.props.icons.Check,
        tooltip: localization.saveTooltip,
        disabled: !isValid, // here we prevent invalid form 
        onClick: this.handleSave
      },
      {
        icon: this.props.icons.Clear,
        tooltip: localization.cancelTooltip,
        onClick: () => {
          this.props.onEditingCanceled(this.props.mode, this.props.data);
        }
      }
    ];
``` 

- Second path is from keyCode enter.

```js
  handleKeyDown = (e) => {
    if (e.keyCode === 13 && e.target.type !== 'textarea') {
      this.handleSave();
    } else if (e.keyCode === 13 && e.target.type === 'textarea' && e.shiftKey) {
      this.handleSave();
    } else if (e.keyCode === 27) {
      this.props.onEditingCanceled(this.props.mode, this.props.data);
    }
  };
```

**To Reproduce**

- Setup a basic validation system
- Write an invalid input
- Use enter to bypass the validation

**Expected behavior**
- Press Enter process must be identical to Check.

**Additional context**
- Issue is available in : mbrn/material-table

**Comment by Domino987** [comment]
The class component is not used anymore.
But the function component will be updated probably today, maybe tomorrow.
Thanks for the catch.

**Comment by Domino987** [comment]
Fixed with 4.3.8
